plugin.rb

```diff
@@ -31,23 +31,58 @@ def self.default_image_url
       "#{Discourse.base_url}/plugins/discourse-patreon-creator/images/patreon-logomark-color-on-white.png"
     end
 
-    def self.store
-      @store ||= PluginStore.new(PLUGIN_NAME)
-    end
-
     def self.get(key)
       store.get(key)
     end
 
+    def self.nsfw_group
+      @nsfw_group = Group.find_by_name(SiteSetting.patreon_creator_nsfw_group)
+    end
+
+    def self.remove(key)
+      store.remove(key)
+    end
+
+    def self.save_user_info(user, email, patreon_id, has_nsfw_campaign)
+      plugin_store_key = "user_info.#{email}"
+      if user
+        user.custom_fields["patreon_id"] = patreon_id
+        user.custom_fields["has_nsfw_campaign"] = has_nsfw_campaign
+        user.save_custom_fields
+
+        # Remove the temp data from the plugin store
+        Patreon.remove(plugin_store_key)
+      else
+        # User has not been created, but they are authorized.
+        # save data in plugin store so we can use in during
+        # This is called during authorization and on the
+        # :user_created callback.
```

Comment:
This comment seems like it either isn't complete or may be two thoughts. 

> so we can use in during This is called 

just reads weird